### PR TITLE
Fixed some of RepeatDBHandling of matched Aggregated Objects ids issues.

### PR DESCRIPTION
**Pull request by tobiasake**
This change solves the storing issues of multiple aggregated objects ids in different MongoDb document indexes and multiple of aggregated objects ids for same subscriptions requirements matched aggregated object ids list in RepeatHandler database collection.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 1]
```diff
@@ -121,8 +121,10 @@ public boolean modifySubscription(Subscription subscription, String subscription
             if (result != null) {
                 String subscriptionIdQuery = String.format(SUBSCRIPTION_ID, subscriptionName);
                 if (!cleanSubscriptionRepeatFlagHandlerDb(subscriptionIdQuery)) {
-                    LOG.error("Failed to clean subscription \"" + subscriptionName
-                            + "\" matched AggregatedObjIds from RepeatFlagHandler database");
+                    LOG.info("Subscription  \"" + subscriptionName
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: From my preivious pull-request, already in master branch. No need to review.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 2]
```diff
@@ -121,8 +121,10 @@ public boolean modifySubscription(Subscription subscription, String subscription
...
+                            + "\" matched aggregated objects id from repeat flag handler database could not be cleaned during the update of the subscription,\n"
+                    		+ "probably due to subscription has never matched any aggregated objects and "
+                            + "no matched aggregated objects id has been stored in database for the specific subscription.");
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: From my preivious pull-request, already in master branch. No need to review.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 3]
```diff
@@ -142,8 +144,10 @@ public boolean deleteSubscription(String subscriptionName) throws AccessExceptio
         if (deleteResult) {
             String subscriptionIdQuery = String.format(SUBSCRIPTION_ID, subscriptionName);
             if (!cleanSubscriptionRepeatFlagHandlerDb(subscriptionIdQuery)) {
-                LOG.error("Failed to clean subscription \"" + subscriptionName
-                        + "\" matched AggregatedObjIds from RepeatFlagHandler database");
+                LOG.info("Subscription  \"" + subscriptionName
+                        + "\" matched aggregated objects id from repeat flag handler database could not be cleaned during the removal of subscription,\n"
+                		+ "probably due to subscription has never matched any aggregated objects and "
+                        + "no matched aggregated objects id has been stored in database for the specific subscription.");
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: From my preivious pull-request, already in master branch. No need to review.

File: src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java [hunk 4]
```diff
@@ -165,7 +165,8 @@ public void in_MongoDb_RepeatFlagHandler_collection_the_subscription_has_matched
      */
     private String getAggregatedObjectId(List<String> resultRepeatFlagHandler, int index) {
         JsonParser parser = new JsonParser();
-        JsonObject jsonObject = parser.parse(resultRepeatFlagHandler.get(index)).getAsJsonObject();
+        JsonObject jsonObject = parser.parse(resultRepeatFlagHandler.get(0)).getAsJsonObject();
+        System.out.println("JSONOBJECT: " + jsonObject.toString());
```
Comment (e-edling-ericsson) [inline src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java]: Should be removed?

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java [hunk 5]
```diff
@@ -66,16 +67,21 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
...
+        LOGGER.debug("AGG OBJECT: " + aggregatedObject);
 
         while (requirementIterator.hasNext()) {
 
             JsonNode aggrObjJsonNode = null;
             ObjectMapper objectMapper = new ObjectMapper();
             try {
                 aggrObjJsonNode = objectMapper.readValue(aggregatedObject, JsonNode.class);
+                
+//                DBObject idObj = (DBObject)aggrObjJsonNode.get("_id").get("$oid");
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: remove out commented code

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java [hunk 6]
```diff
@@ -66,16 +67,21 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
...
+                LOGGER.debug("AGGREGATED OBJECT ID: " + aggrObjJsonNode.asText());
             } catch (Exception e) {
                 LOGGER.error(e.getMessage(), e);
             }
+            LOGGER.debug("1111111111111111111111111111111111111111111111111111111111111111111111111111111");
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: Remove? :)

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java [hunk 7]
```diff
@@ -91,7 +97,9 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
                         + subscriptionName + "\nand has Subscrption Repeat flag set to: " + subscriptionRepeatFlag);
                 break;
             }
-
+            
+            LOGGER.debug("222222222222222222222222222222222222222222222222222222222222222222222222222222222222");
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: Remove? :)

File: src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java (hunk 4 above)
Comment (tobiasake) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java]: fixed.

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java (hunk 5 above)
Comment (tobiasake) [reply src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: fixed.

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java (hunk 6 above)
Comment (tobiasake) [reply src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: firxed.

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java (hunk 7 above)
Comment (tobiasake) [reply src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: fixed.

File: src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java [hunk 8]
```diff
@@ -64,19 +65,10 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
...
         while (requirementIterator.hasNext()) {
 
-            JsonNode aggrObjJsonNode = null;
-            ObjectMapper objectMapper = new ObjectMapper();
-            try {
-                aggrObjJsonNode = objectMapper.readValue(aggregatedObject, JsonNode.class);
-            } catch (Exception e) {
-                LOGGER.error(e.getMessage(), e);
-            }
-
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/subscriptionhandler/RunSubscription.java]: "aggrObjJsonNode" object was never used in the code below, so removed this unused code section.

**Pull request merged**